ldclient/twisted_client.py

```diff
@@ -0,0 +1,80 @@
+from functools import partial
+
+from twisted.internet import defer
+from twisted.internet.defer import DeferredList
+
+from ldclient import LDClient
+from ldclient import log
+from ldclient.flag import _get_variation, _evaluate_index, _get_off_variation
+
+
+class TwistedLDClient(LDClient):
+    @defer.inlineCallbacks
+    def _evaluate_and_send_events(self, flag, user, default):
+        value = yield self._evaluate(flag, user)
+        if value is None:
+            value = default
+        log.info("value: " + str(value))
+        self._send_event({'kind': 'feature', 'key': flag.get('key'), 'user': user, 'value': value,
+                          'default': default, 'version': flag.get('version')})
+        defer.returnValue(value)
+
+    def _evaluate(self, flag, user):
+        if flag.get('on', False):
+            def cb(result):
+                if result is not None:
+                    return result
+                return _get_off_variation(flag)
+
+            value = self._evaluate_internal(flag, user)
+            value.addBoth(cb)
+            return value
+
+        return _get_off_variation(flag)
+
+    def _evaluate_internal(self, flag, user):
+        def check_prereq_results(result):
+            prereq_ok = True
+            for r in result:  # r is a tuple of 2 booleans: (error, prereqMatches)
```

Comment:
Can we use [NamedTuples](https://docs.python.org/2/library/collections.html#collections.namedtuple) to increase the readability?